third_party/move/move-vm/types/src/code_storage/interface.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use bytes::Bytes;
+use move_binary_format::errors::VMResult;
+use move_core_types::{account_address::AccountAddress, identifier::IdentStr};
+
+/// Storage that contains serialized modules. Clients can implement this trait for their own
+/// backends, so that [ModuleStorage] can be built on top of it.
+pub trait ModuleBytesStorage {
```

Comment:
This seems to be a duplicate of move_vm_types::ModuleResolver?

Reply:
The plan is to remove ModuleResolver altogether because we should not resolve just bytes. I used a new trait to differentiate between V1 and V2, and avoid using it for both flows. This also helps to avoid data coupling because we will not need MoveResolver, there is just a ResourceResolver for data, and ModuleStorage for code.